src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -106,49 +102,32 @@ public <T> T unwrap(Class<T> unwrapType) {
     }
 
     @Override
-    public void configure(Map<String, Object> configValues) {
-        var jdbcUrl = configValues.get(JAKARTA_JDBC_URL);
+    public void configure(Map<String, Object> configProperties) {
+        var jdbcUrl = configProperties.get(JAKARTA_JDBC_URL);
 
-        if (mongoClientCustomizer == null && jdbcUrl == null) {
+        if (mongoDialectConfigurator == null && jdbcUrl == null) {
             throw new HibernateException(format(
                     "Configuration property [%s] is required unless %s is provided",
-                    JAKARTA_JDBC_URL, MongoClientCustomizer.class.getName()));
+                    JAKARTA_JDBC_URL, MongoDialectConfigurator.class.getName()));
         }
 
-        var connectionString = jdbcUrl == null ? null : getConnectionString(jdbcUrl);
-
-        final MongoClientSettings.Builder clientSettingsBuilder;
-        if (mongoClientCustomizer != null) {
-            clientSettingsBuilder = MongoClientSettings.builder();
-            mongoClientCustomizer.customize(clientSettingsBuilder, connectionString);
-        } else {
-            clientSettingsBuilder = MongoClientSettings.builder().applyConnectionString(connectionString);
+        try {
+            MongoDialectSettings.Builder mongoDialectSettingsBuilder = MongoDialectSettings.builder(configProperties);
+            if (mongoDialectConfigurator != null) {
+                mongoDialectConfigurator.configure(mongoDialectSettingsBuilder);
```

Comment:
Note that there is no longer a way for an application to decide whether to apply the connection string (if any) before or after doing other configuration changes, because we apply it. If an application absolutely has to apply its connection string after doing some other configuration changes, then it can omit configuring `JAKARTA_JDBC_URL`, and apply whatever connections string it needs to how it wants that via `MongoDialectConfigurator`.